### PR TITLE
feat(#680): added more javadoc

### LINKED ISSUE #680: Have Swagger docs and Javadocs generated?
Given the OpenSSF certification we need to: "The project MUST provide reference documentation that describes the external interface (both input and output) of the software produced by the project. [documentation_interface] ". For this we should implement the generation of Swagger and Java docs.
These docs don't have to be stored: they can be a result of a github action where people can retrieve them if required.

### PR DESCRIPTION
**Pull request by commjoen**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [ ] A new challenge
-   [x] Additional documentation
-   [ ] Something else

### Relations

Closes #680

### Checklist:

-   [ ] All the contributions made are solely the work of me and my co-authors
-   [ ] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: src/main/java/org/owasp/wrongsecrets/HerokuWebSecurityConfig.java [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import org.springframework.security.web.PortMapper;
 import org.springframework.security.web.SecurityFilterChain;
 
+/**
+ * Used to implement to https redirect for our Heroku hosted workload.
```
Comment (bendehaan) [inline src/main/java/org/owasp/wrongsecrets/HerokuWebSecurityConfig.java:12]:
```suggestion
 * Used to implement https redirect for our Heroku-hosted workload.
```

File: src/main/java/org/owasp/wrongsecrets/InMemoryScoreCard.java [hunk 2]
```diff
@@ -5,6 +5,9 @@
 import java.util.HashSet;
 import java.util.Set;
 
+/**
+ * In memory implementation of the ScoreCard (E.g. no persistence).
```
Comment (bendehaan) [inline src/main/java/org/owasp/wrongsecrets/InMemoryScoreCard.java:9]:
```suggestion
 * In-memory implementation of the ScoreCard (E.g. no persistence).
```

File: src/main/java/org/owasp/wrongsecrets/ScoreCard.java [hunk 3]
```diff
@@ -2,14 +2,39 @@
 
 import org.owasp.wrongsecrets.challenges.Challenge;
 
+/**
+ * Interface of a scorecard where progress of a player is stored into.
```
Comment (bendehaan) [inline src/main/java/org/owasp/wrongsecrets/ScoreCard.java:6]:
```suggestion
 * Interface of a scorecard where a player's progress is stored into.
```

**Comment by commjoen** [comment]
@bendehaan all changes applied :D

**Review by bendehaan — APPROVED** [review]

**Pull request merged**